huaweicloud/services/dew/data_source_huaweicloud_csms_events.go

```diff
@@ -0,0 +1,204 @@
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"notification": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceDewCsmsEventsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		listEventsHttpUrl = "v1/{project_id}/csms/events"
+		listEventsProduct = "kms"
+	)
+	listEventsClient, err := cfg.NewServiceClient(listEventsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating KMS client: %s", err)
+	}
+
+	listEventsPath := listEventsClient.Endpoint + listEventsHttpUrl
+	listEventsPath = strings.ReplaceAll(listEventsPath, "{project_id}", listEventsClient.ProjectID)
+
+	listEventsResp, err := pagination.ListAllItems(
+		listEventsClient,
+		"marker",
+		listEventsPath,
+		&pagination.QueryOpts{MarkerField: "name"})
+
+	if err != nil {
+		return diag.Errorf("error retrieving CSMS events: %s", err)
+	}
+
+	listEventsRespJson, err := json.Marshal(listEventsResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var listEventsRespBody interface{}
+	err = json.Unmarshal(listEventsRespJson, &listEventsRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("events", filterListEventsBody(
+			flattenListEventsBody(listEventsRespBody), d)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenListEventsBody(resp interface{}) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	curJson := utils.PathSearch("events", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		createAt := utils.PathSearch("create_time", v, 0)
+		updateAt := utils.PathSearch("update_time", v, 0)
+		rst = append(rst, map[string]interface{}{
+			"name":         utils.PathSearch("name", v, nil),
+			"event_id":     utils.PathSearch("event_id", v, nil),
+			"event_types":  utils.PathSearch("event_types", v, nil),
+			"status":       utils.PathSearch("state", v, nil),
+			"created_at":   utils.FormatTimeStampRFC3339(int64(createAt.(float64))/1000, false),
+			"updated_at":   utils.FormatTimeStampRFC3339(int64(updateAt.(float64))/1000, false),
+			"notification": utils.PathSearch("notification", v, nil),
+		})
+	}
+	return rst
+}
+
+func filterListEventsBody(all []interface{}, d *schema.ResourceData) []interface{} {
```

Comment:
There may be risks when filtering both name and status simultaneously, you can try the method of d.GetOk()